src/youtube-player/fake-youtube-player.ts

```diff
@@ -45,7 +45,7 @@ export function createFakeYtNamespace(): FakeYtNamespace {
 
       for (const [event, callback] of playerSpy.addEventListener.calls.allArgs()) {
         if (event === name) {
-          callback(arg);
+          callback(arg as YT.PlayerEvent);
```

Comment:
Note: I tried fixing this by improving type-safety in this method overall, but it looks like this is not possible as the passed `arg` needs to be less tightened. Also the `as any` cast above can't be replaced with `YT.PlayerEvent`.

This test utility seems to need a general cleanup for type safety, but out of scope for this PR. It just caused a compilation failure as the `addEventListener` callback type is now inferred properly and expects `YT.PlayerEvent`.